.github/workflows/python_cibuildwheel.yml

```diff
@@ -12,173 +12,29 @@ on:
-            arch: x64
-            allow_failure: false
-
-          # Window 32 bit
-          - os: windows-2019
-            python: 38
-            bitness: 32
-            platform_id: win32
-            cmake_compiler: vc16
-            arch: x86
-            allow_failure: true
-          - os: windows-latest
-            python: 39
-            bitness: 32
-            platform_id: win32
-            cmake_compiler: vc17
-            arch: x86
-            allow_failure: true
-
-          # Linux 64 bit manylinux2014
-          - os: ubuntu-latest
-            python: 37
-            bitness: 64
-            platform_id: manylinux_x86_64
-            manylinux_image: manylinux2014
-            cmake_compiler: default
-            allow_failure: false
-          - os: ubuntu-latest
-            python: 38
-            bitness: 64
-            platform_id: manylinux_x86_64
-            manylinux_image: manylinux2014
-            cmake_compiler: default
-            allow_failure: false
-          - os: ubuntu-latest
-            python: 39
-            bitness: 64
-            platform_id: manylinux_x86_64
-            manylinux_image: manylinux2014
-            cmake_compiler: default
-            allow_failure: false
-          - os: ubuntu-latest
-            python: 310
-            bitness: 64
-            platform_id: manylinux_x86_64
-            manylinux_image: manylinux2014
-            cmake_compiler: default
-            allow_failure: false
-
-          # MacOS x86_64
-          - os: macos-latest
-            bitness: 64
-            python: 37
-            platform_id: macosx_x86_64
-            cmake_compiler: default
-            allow_failure: false
-          - os: macos-latest
-            bitness: 64
-            python: 38
-            platform_id: macosx_x86_64
-            cmake_compiler: default
-            allow_failure: false
-          - os: macos-latest
-            bitness: 64
-            python: 39
-            platform_id: macosx_x86_64
-            cmake_compiler: default
-            allow_failure: false
-          - os: macos-latest
-            bitness: 64
-            python: 310
-            platform_id: macosx_x86_64
-            cmake_compiler: default
-            allow_failure: false
-
-          # MacOS arm64
-          - os: macos-latest
-            bitness: 64
-            python: 38
-            platform_id: macosx_arm64
-            cmake_compiler: default
-            allow_failure: false
-          - os: macos-latest
-            bitness: 64
-            python: 39
-            platform_id: macosx_arm64
-            cmake_compiler: default
-            allow_failure: false
-          - os: macos-latest
-            bitness: 64
-            python: 310
-            platform_id: macosx_arm64
-            cmake_compiler: default
-            allow_failure: false
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: [36, 37, 38, 39, 310]
```

Comment:
I'm still breaking it up by python version within a given OS or it is super slow, cf commit message in a2cd8b4c572bb445e7d65ff6713638193dcc9270 

> About 1h15 on mac, 45 min on windows